src/librustc/lib.rs

```diff
@@ -60,7 +60,7 @@
 #![cfg_attr(stage0, feature(const_fn))]
 #![cfg_attr(not(stage0), feature(const_atomic_bool_new))]
 
-#![recursion_limit="256"]
+#![recursion_limit="512"]
```

Comment:
Is this needed with the new shortcutting?

Reply:
This is due to the increased amount of queries.

Reply:
Shouldn't the shortcutting avoid these?

Reply:
I mean the macros which defines the queries causes the overflow. Not related to immovable types.